### PR TITLE
[EventHubs] LastEnqueuedOffset partition property changed from 'string' to 'int'

### PR DESCRIPTION
**Pull request by kinelski**
Partition Properties' offsets are supposed to assume numeric values: -1 when no event information is stored; or a non-negative integer number otherwise.

Tests and other parts of the source code were updated accordingly to accommodate the changes.

Related issue: #6779

### CODE REVIEW

**Review by jsquire — APPROVED** [review]
Looks good to me.   I left a couple of small comments, nothing that I'd consider blocking.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Compatibility/TrackOneEventHubClient.cs [hunk 1]
```diff
@@ -203,7 +203,7 @@ public override async Task<EventHubProperties> GetPropertiesAsync(CancellationTo
                 runtimeInformation.PartitionId,
                 runtimeInformation.BeginSequenceNumber,
                 runtimeInformation.LastEnqueuedSequenceNumber,
-                runtimeInformation.LastEnqueuedOffset,
+                int.Parse(runtimeInformation.LastEnqueuedOffset),
```
[the remark this answers is not in the document]
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Compatibility/TrackOneEventHubClient.cs]: I have added a `TryParse` just to be sure. It's better to be safe than sorry, but I think it won't fail as long as we don't do anything wrong in the internal code.

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneEventHubClientTests.cs [hunk 2]
```diff
@@ -538,7 +538,11 @@ protected override PartitionReceiver OnCreateReceiver(string consumerGroupName,
             protected override Task<EventHubPartitionRuntimeInformation> OnGetPartitionRuntimeInformationAsync(string partitionId)
             {
                 GetPartitionRuntimePartitionInvokedWith = partitionId;
-                return Task.FromResult(new EventHubPartitionRuntimeInformation());
+
+                var ehPartitionRuntimeInformation = new EventHubPartitionRuntimeInformation();
```
[the remark this answers is not in the document]
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneEventHubClientTests.cs]: Agreed. Removed it!

**Comment by jsquire** [comment]
:shipit:

**Pull request merged**